### PR TITLE
DO NOT MERGE: generating .asciidoc manpages and pushing to machinekit-man

### PR DESCRIPTION
**Pull request by luminize**
I've just transferred the `machinekit-man` to the Machinekit organisation.
This branch has the build target for generating the `.asciidoc `files.
For that to happen I needed the `.gitignore` to see the `.asciidoc `files.

I'll elaborate in #433

### CODE REVIEW

**Comment by zultron** [comment]
Hi @luminize,

I'm trying to understand what this PR is about.  It's submitted against the main `machinekit` repo, but aren't these `.asciidoc` files meant for the `machinekit-man` repo?

What does a `.gitignore` file with the single entry `.gitignore` do, such as [this one][1]?  Is it functionally different from no `.gitignore` file at all?

[1]: https://github.com/luminize/machinekit/tree/machinekit-man/man/man-generated/hal-components

In the `Makefile`, it's possible to create functions that dedup the copy-and-paste repetition if you like, of course completely optional.

Sorry if I'm missing something obvious.  My attention is split several ways at the moment.  ;)

**Comment by luminize** [comment]
Hi @zultron,

I made this PR against `machinekit` to (1) see if you want to fiddle with pushing generated `.asciidoc` files to an external repo by the buildbot and (2) for me to see if my makefile doesn't introduce errors. I'm in no way trying to push you or anything. The `machinekit-man` repo was where we and @mhaberler talked about recently. Pretty greenish regarding "right" way to do this. Learning as i go.

I'm still working on the manpage -> asciidoc so I'm perfectly OK to close this and re-open if I want the tickle the buildbot.

As for your question about the empty `.gitignore` files. Those are for making sure there is a directory where I can put the generated files. Without something in a directory `git` just removes the directory.
But I'm sure there is a better way to not get a fault while compiling when the directory for the target is missing.

Another thing, If in the future the buildbot wants to push generated `.asciidoc` files to a git repo. There must be a way to commit (and thus see with a `.gitignore` not ignoring line) the changed/updated `.asciidoc` files to be able to commit to the local repo. -few-

long story short. I'm "goofing" around for getting something smart generated-and-pushable... Don't know if this makes any sense :)

**Comment by zultron** [comment]
Ah this all makes sense!

I would indeed like to fiddle with the Buildbot pushing to a repo.  No time right away.

You probably know, you can also force-push your `machinekit-man` branch to re-tickle the Buildbot without closing/opening tickets.

The `.gitignore` trick works, or as in other places in the code like `src/rtapi/Submakefile`, a `@mkdir -p $(SOMEDIR)` command is inserted into the recipe.

I've been thinking about how to deal with a parent git repo wishing to ignore e.g. these generated `.asciidoc` files, but the subtree wishing not to ignore them.  No bright ideas yet without really understanding `git subtree` and some experimentation....

The problem with becoming the expert on a topic is you find there's nobody who can help you anymore.  Sorry, Bas.  ;)

**Comment by luminize** [comment]
> On 19 Jan 2015, at 22:34, John Morris <notifications@github.com> wrote:
> 
> No bright ideas yet without really understanding git subtree and some experimentation....
> 
> The problem with becoming the expert on a topic is you find there's nobody who can help you anymore.  Sorry, Bas. ;)
> 

:) no worries
experimentation is part of life and frequent backing up is the remedy.

For now I have this branch (of this PR), with only git not ignoring in the last few commits. [Another branch](https://github.com/luminize/machinekit/tree/man-generated) is used to change scripts and Makefile, and I then rebase this branch with that one. =

**Comment by luminize** [comment]
I close this PR since testing has been done. generated asciidocs have no place in machinekit so the PR #462 replaces this one
